sdk/src/wallet/operations/transaction/account.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::{
+        address::Address,
+        output::{
+            feature::{BlockIssuerFeature, BlockIssuerKey, BlockIssuerKeys, Ed25519BlockIssuerKey},
+            unlock_condition::AddressUnlockCondition,
+            AccountId, AccountOutput, OutputId,
+        },
+    },
+    wallet::{
+        operations::transaction::{TransactionOptions, TransactionWithMetadata},
+        Error, Result, Wallet,
+    },
+};
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    /// Transitions an implicit account to an account.
+    pub async fn implicit_account_transition(&self, output_id: &OutputId) -> Result<TransactionWithMetadata> {
+        self.sign_and_submit_transaction(self.prepare_implicit_account_transition(output_id).await?, None)
+            .await
+    }
+
+    /// Prepares to transition an implicit account to an account.
+    pub async fn prepare_implicit_account_transition(&self, output_id: &OutputId) -> Result<PreparedTransactionData> {
+        let implicit_account_data = self.data().await.unspent_outputs.get(output_id).cloned();
+
+        let implicit_account = if let Some(implicit_account_data) = &implicit_account_data {
+            if implicit_account_data.output.is_implicit_account() {
+                implicit_account_data.output.as_basic()
+            } else {
+                return Err(Error::ImplicitAccountNotFound);
+            }
+        } else {
+            return Err(Error::ImplicitAccountNotFound);
+        };
+
+        let public_key = if let Some(bip_path) = self.bip_path().await {
```

Comment:
I think the user should be able to provide the public key to the function as parameter

Reply:
Probably somewhat related to the TODO issue in the else case

Reply:
I guess I agree but there are more combinations we can do, I have opened an issue for it